internal/cli/opsmanager/serverusage/download.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serverusage
+
+import (
+	"fmt"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/ops-manager/opsmngr"
+)
+
+type DownloadOpts struct {
+	cli.DownloaderOpts
+	startDate string
+	endDate   string
+	format    string
+	store     store.ServerUsageReportDownloader
+}
+
+var downloadMessage = "Download of %s completed.\n"
+
+func (opts *DownloadOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *DownloadOpts) Run() error {
+	f, err := opts.NewWriteCloser()
+	if err != nil {
+		return err
+	}
+
+	if err := opts.store.DownloadServerUsageReport(opts.newServerTypeOptions(), f); err != nil {
+		_ = opts.OnError(f)
+		return err
+	}
+	if opts.Out != "/dev/stdout" {
+		fmt.Printf(downloadMessage, opts.Out)
+	}
+	return f.Close()
+}
+
+func (opts *DownloadOpts) newServerTypeOptions() *opsmngr.ServerTypeOptions {
+	return &opsmngr.ServerTypeOptions{
+		StartDate: opts.startDate,
+		EndDate:   opts.endDate,
+	}
+}
+
+func (opts *DownloadOpts) initDefaultOut() {
+	if opts.Out == "" {
+		opts.Out = "serverUsageReport" + opts.format
```

Comment:
how does this format works? the docs say that it always returns a tgz, https://docs.opsmanager.mongodb.com/current/reference/api/diagnostics/get-project-diagnostic-archive/

Reply:
I think that I may have used the wrong end point. Here I have used https://docs.opsmanager.mongodb.com/current/reference/api/usage/create-one-report/ which is the one that I have implemented. Can you confirm that I have used the wrong endpoint?

Reply:
no, you used the right one but that create us confusing, one thing here, for that endpoint the format is a query param, which you're not passing as a query param (newServerTypeOptions)

Reply:
I have created a [docs](https://jira.mongodb.org/browse/DOCSP-12840) ticket to update that documentation